src/providers/aws.rs

```diff
@@ -1290,23 +1298,34 @@ impl RegionLauncher {
         tracing::debug!("cleaning up temporary resources");
         if !self.ssh_key_name.trim().is_empty() {
             let key_span = tracing::trace_span!("key", name = %self.ssh_key_name);
-            async {
+            let r = async {
                 tracing::trace!("removing keypair");
                 let mut req = rusoto_ec2::DeleteKeyPairRequest::default();
                 req.key_name = self.ssh_key_name.clone();
                 if let Err(e) = client.delete_key_pair(req).await {
                     tracing::warn!("failed to clean up temporary SSH key: {}", e);
+                    Err(e).context("failed to clean up temporary SSH key")?;
+                    unreachable!();
                 }
+
+                Ok(())
             }
             .instrument(key_span)
             .await;
+
+            if r.is_err() {
```

Comment:
Here too I think we can just early-return on error with `?`. Which should avoid the need for `res`.